apm-agent-plugins/apm-log-shader-plugin/apm-log4j1-plugin/src/main/java/co/elastic/apm/agent/log4j1/Log4J1EcsReformattingHelper.java

```diff
@@ -50,10 +51,17 @@ protected String getAppenderName(WriterAppender appender) {
     }
 
     @Override
-    protected Layout createEcsFormatter(String eventDataset, @Nullable String serviceName) {
+    protected Layout createEcsFormatter(String eventDataset, @Nullable String serviceName, @Nullable String serviceNodeName,
+                                        @Nullable Map<String, String> additionalFields, Layout originalFormatter) {
         EcsLayout ecsLayout = new EcsLayout();
         ecsLayout.setServiceName(serviceName);
+        ecsLayout.setServiceNodeName(serviceNodeName);
         ecsLayout.setEventDataset(eventDataset);
+        if (additionalFields != null) {
+            for (Map.Entry<String, String> keyValuePair : additionalFields.entrySet()) {
+                ecsLayout.setAdditionalField(keyValuePair.getKey() + "=" + keyValuePair.getValue());
+            }
+        }
```

Comment:
[minor] there are multiple variants of this, might be worth having a common code to add extra fields in `AbstractEcsReformattingHelper` to avoid duplication.

Reply:
There are multiple variants because the way to add additional data is library-specific, in Logback and log4j2 it even requires a library-specific type for the added field.